### ISSUE TITLE
Allow more minimal graphviz representation

### ISSUE THREAD

**Issue by drbenvincent** [pymc-devs/pymc#6716]
In Bayesian Network and Causal Graph land, you will often see quite minimal networks which focus on the core variables of interest and their statistical and/or causal relations. Here is an example.
![Screenshot 2023-05-12 at 15 45 10](https://github.com/pymc-devs/pymc/assets/6765047/4e81a2b7-2e9f-416c-9e07-765bed775d9d)

If you code up such a model in PyMC (e.g. with linear relationships between the nodes) you get a 'full' representation of the graph that PyMC builds, like this...

![Screenshot 2023-05-12 at 15 49 06](https://github.com/pymc-devs/pymc/assets/6765047/ef3bc582-486e-4c0c-b30b-7bbbd46d101b)

But it would be extremely useful to be able to produce more minimal representations, like the first figure.

So this is a feature request for that. 

In terms of implementation, it could be done using a `minimal=True` kwarg to `pm.model_to_graphviz`. Alternatively it could be a new function, such as `pm.model_to_simple_graphviz`. My initial preference would be for the latter because you could imaging wanting to have a number of other kwargs to control the visualisation which you might not want in the existing function.

This issue is not entirely unrelated to #5794, but it is not the same

**Comment by ricardoV94** [comment]
I suspect that the `minimal=True` is a bit ambiguous and different users may want different "minimal" versions of the graph. I think supporting `var_names` would be the most flexible way and hopefully not terrible to work with.

**Comment by drbenvincent** [comment]
Though having to manually specify could be far from the quick and easy process of visualising that we have at the moment.

**Comment by drbenvincent** [comment]
Perhaps `style = "causal_dag"`. That would be future proof and allow for multiple different graphviz DAG styles.

**Comment by fonnesbeck** [comment]
Perhaps something like `include_dist=False`, which would exclude the tilde and distribution type from the graph?

**Comment by drbenvincent** [comment]
> Perhaps something like `include_dist=False`, which would exclude the tilde and distribution type from the graph?

That would be pretty good. Presumably that alone would still leave the RV's and deterministics in the graph? But I like the idea in general of allowing users to configure the graphviz output. I think this is a really strong aspect of PyMC and that there are things we can do to make the outputs more suited to different contexts.

**Comment by larryshamalama** [comment]
I think that there is some ambiguity in identifying variables that are necessary for display in the context of a causal DAG. In the graph above, it seems that variables that we are interested in are 1) observed MeasurableVariables or 2) MutableDatas?

So far, I feel like adding the option `include_dist` (combined with the use of `var_names`) could work well, but I'm interested to hear what other ideas we may have

**Comment by drbenvincent** [comment]
Agreed. It's not necessarily trivial to work out what the core variables are, so I'm definitely coming around to this idea. Episode N of @ricardoV94 is correct.

**Comment by ColCarroll** [comment]
When I want a particular output, I'll often `print(pm.model_to_graphviz(model))`, and just use that as a starting point for custom graphviz.

![image](https://github.com/pymc-devs/pymc/assets/2295568/7be8e057-5a7c-4485-a984-3a078084e2d4)
